packages/overlay/test/overlay.test.ts

```diff
@@ -0,0 +1,231 @@
+        if (outerPopover.parentElement) {
+            expect(outerPopover.parentElement.id).not.to.equal(
+                'overlay-content'
+            );
+        }
+        expect(isVisible(outerPopover)).to.be.true;
+    });
+
+    it('opens hover overlay', async () => {
+        const button = testDiv.querySelector('#first-button') as HTMLElement;
+        const hoverOverlay = testDiv.querySelector('#hover-1') as HTMLElement;
+        const clickOverlay = testDiv.querySelector(
+            '#outer-popover'
+        ) as HTMLElement;
+
+        expect(button).to.exist;
+        expect(hoverOverlay).to.exist;
+
+        expect(isVisible(hoverOverlay)).to.be.false;
+        expect(isVisible(clickOverlay)).to.be.false;
+
+        Overlay.open(button, 'hover', hoverOverlay, {
+            delayed: false,
+            placement: 'top',
+            offset: 10,
+        });
+
+        // Wait for the DOM node to be stolen and reparented into the overlay
+        await waitForPredicate(
+            () =>
+                !!(
+                    hoverOverlay.parentElement &&
+                    hoverOverlay.parentElement.id !== 'overlay-content'
+                )
+        );
+
+        expect(hoverOverlay.parentElement).to.exist;
+        if (hoverOverlay.parentElement) {
+            expect(hoverOverlay.parentElement.id).not.to.equal(
+                'overlay-content'
+            );
+        }
+        expect(isVisible(hoverOverlay)).to.be.true;
+
+        // Opening click overlay should close the hover overlay
+        Overlay.open(button, 'click', clickOverlay, {
+            delayed: false,
+            placement: 'bottom',
+            offset: 10,
+        });
+
+        // Wait for the DOM node to be stolen and reparented into the overlay
+        await waitForPredicate(
+            () =>
+                !!(
+                    clickOverlay.parentElement &&
+                    clickOverlay.parentElement.id !== 'overlay-content' &&
+                    hoverOverlay.parentElement &&
+                    hoverOverlay.parentElement.id === 'overlay-content'
+                )
+        );
+
+        if (hoverOverlay.parentElement) {
+            expect(hoverOverlay.parentElement.id).to.equal('overlay-content');
+        }
+
+        expect(isVisible(hoverOverlay)).to.be.false;
+        expect(isVisible(clickOverlay)).to.be.true;
+    });
+
+    it('opens custom overlay', async () => {
+        const button = testDiv.querySelector('#first-button') as HTMLElement;
+        const customOverlay = testDiv.querySelector('#hover-1') as HTMLElement;
+        const clickOverlay = testDiv.querySelector(
+            '#outer-popover'
+        ) as HTMLElement;
+
+        expect(button).to.exist;
+        expect(customOverlay).to.exist;
+
+        expect(isVisible(customOverlay)).to.be.false;
+        expect(isVisible(clickOverlay)).to.be.false;
+
+        Overlay.open(button, 'custom', customOverlay, {
```

Comment:
Does the ability to do this fix #321? If so, can you add that to the description so it auto closes on merge. If not, could you add some information about next steps to that issue?

Reply:
I think this solves #321. If we hit other issues we can follow up in a later issue. I added the issue to the description for this PR.